blockchain/tx_list.go

```diff
@@ -281,14 +312,24 @@ func (l *txList) Add(tx *types.Transaction, priceBump uint64) (bool, *types.Tran
 	// If there's an older better transaction, abort
 	old := l.txs.Get(tx.Nonce())
 	if old != nil {
-		if tx.Type().IsCancelTransaction() {
-			logger.Trace("New tx is a cancel transaction. replace it!", "old", old.String(), "new", tx.String())
-		} else {
+		// TODO : Need to KIP-71 hardfork.
+		//if tx.Type().IsCancelTransaction() {
+		//	logger.Trace("New tx is a cancel transaction. replace it!", "old", old.String(), "new", tx.String())
+		//} else {
+		//	logger.Trace("already nonce exist", "nonce", tx.Nonce(), "with gasprice", old.GasPrice(), "priceBump", priceBump, "new tx.gasprice", tx.GasPrice())
+		//	return false, nil
+		//}
+
+		// If gas price of older is bigger than newer, abort.
+		if old.GasPrice().Cmp(tx.GasPrice()) >= 0 {
 			logger.Trace("already nonce exist", "nonce", tx.Nonce(), "with gasprice", old.GasPrice(), "priceBump", priceBump, "new tx.gasprice", tx.GasPrice())
 			return false, nil
+		} else {
+			// Otherwise overwrite the old transaction with the current one
+			logger.Trace("Gas price of new tx is bigger than older tx!", "old", old.String(), "new", tx.String())
```

Comment:
Please see above comments.

Reply:
What do you mean? I'm pointing out the semantics of the logging. The new tx will substitute the older tx if the gas price is higher which is verified with that logging.

Reply:
I just intended to maintain an existed code and seperate logic before and after hardfork.
It will be uncommented and modified after merging governance related PR.

Reply:
The log is already uncommented and was created by your change too. Moreover, the semantics of the log would be preserved because it definitely substitutes the older tx. Very confusing.

Reply:
Sorry.. I just confused. Your suggestion is more correctly. Thanks for review.